compiler/aarch64/codegen/UnaryEvaluator.cpp

```diff
@@ -20,31 +20,38 @@
  *******************************************************************************/
 
 #include "codegen/CodeGenerator.hpp"
+#include "codegen/GenerateInstructions.hpp"
 #include "codegen/TreeEvaluator.hpp"
+#include "il/Node.hpp"
+#include "il/Node_inlines.hpp"
+
+// Common evaluator for integer xconst
+static TR::Register *commonConstEvaluator(TR::Node *node, int32_t value, TR::CodeGenerator *cg)
+   {
+   TR::Register *tempReg = node->setRegister(cg->allocateRegister());
```

Comment:
This isn't an issue here (I think), but you generally `setRegister` as one of the last operations in an evaluator (not the first).  This is because it actually has operations to set the liveness of the virtual register (which affects things like its interference with other registers) and other subtle effects that escape me off the top of my head.  Since these are consts and don't have any children to evaluate this will work out in the end.  But just a note of caution when you `setRegister` for other kinds of nodes.